cmd/renterd/main.go

```diff
@@ -58,117 +113,121 @@ func check(context string, err error) {
 	}
 }
 
-func getAPIPassword() string {
-	if apiPassword == nil {
-		pw := os.Getenv("RENTERD_API_PASSWORD")
-		if pw != "" {
-			fmt.Println("Using RENTERD_API_PASSWORD environment variable.")
-			apiPassword = &pw
-		} else {
-			fmt.Print("Enter API password: ")
-			pw, err := term.ReadPassword(int(os.Stdin.Fd()))
-			fmt.Println()
-			if err != nil {
-				log.Fatal(err)
-			}
-			s := string(pw)
-			apiPassword = &s
-		}
+func mustLoadAPIPassword() {
+	if len(cfg.HTTP.Password) != 0 {
+		return
 	}
-	return *apiPassword
-}
 
-func getSeed() types.PrivateKey {
-	if seed == nil {
-		phrase := os.Getenv("RENTERD_SEED")
-		if phrase != "" {
-			fmt.Println("Using RENTERD_SEED environment variable")
-		} else {
-			fmt.Print("Enter seed: ")
-			pw, err := term.ReadPassword(int(os.Stdin.Fd()))
-			check("Could not read seed phrase:", err)
-			fmt.Println()
-			phrase = string(pw)
-		}
-		key, err := wallet.KeyFromPhrase(phrase)
-		if err != nil {
-			log.Fatal(err)
-		}
-		seed = &key
+	fmt.Print("Enter API password: ")
+	pw, err := term.ReadPassword(int(os.Stdin.Fd()))
+	fmt.Println()
+	if err != nil {
+		log.Fatal(err)
 	}
-	return *seed
+	cfg.HTTP.Password = string(pw)
 }
 
-func parseEnvVar(s string, v interface{}) {
-	if env, ok := os.LookupEnv(s); ok {
-		if _, err := fmt.Sscan(env, v); err != nil {
-			log.Fatalf("failed to parse %s: %v", s, err)
-		}
-		fmt.Printf("Using %s environment variable\n", s)
+func mustLoadRecoveryPhrase() {
+	phrase := cfg.Seed
+	if len(phrase) == 0 {
```

Comment:
This breaks nodes which only have the autopilot enabled. Because right now they are not required to have the seed specified.